java/client/src/main/java/glide/api/BaseClient.java

```diff
@@ -632,9 +633,24 @@ public CompletableFuture<GlideString[]> mget(@NonNull GlideString[] keys) {
     }
 
     @Override
-    public CompletableFuture<String> mset(@NonNull Map<String, String> keyValueMap) {
-        String[] args = convertMapToKeyValueStringArray(keyValueMap);
-        return commandManager.submitNewCommand(MSet, args, this::handleStringResponse);
+    public <ArgType> CompletableFuture<String> mset(@NonNull Map<ArgType, ArgType> keyValueMap) {
+        if (keyValueMap.isEmpty()) {
+            throw new IllegalArgumentException("empty map");
+        }
+
```

Comment:
You can omit this. We have tests where intentionally send to redis an empty map

Reply:
Not sure I follow. This isn't for testing purpose, if we remove this and get an empty map with mset we'll recieve 
NoSuchElementException so isn't his necessary?

Reply:
if map is empty - submit command with 0 args. But track arg type to avoid issues.

Reply:
@Yury-Fridlyand  so basically you want the below code to be enclosed within:

```java

if (!keyValueMap.isEmpty()) {
  ArgType firstValue = keyValueMap.keySet().iterator().next();
  ..
} else {
  // submit empty array
  return commandManager.submitNewCommand(MSet, new String[] {}, this::handleStringResponse);
}

```

Reply:
yes